stumpy/scrump.py

```diff
@@ -106,6 +110,12 @@ def _compute_PI(
             zone_start = max(0, i - excl_zone)
             zone_stop = min(l, i + excl_zone)
             squared_distance_profile[zone_start : zone_stop + 1] = np.inf
+
+            # only for self-join
+            mask = squared_distance_profile < P_squared[thread_idx]
+            P_squared[thread_idx][mask] = squared_distance_profile[mask]
```

Comment:
I think it can. CUDA document says that some of advance indexing is supported. So, since `P_squared[thread_idx][mask]` works, I assume the  `P_squared[thread_idx, mask]` should work as well.

Reply:
Note that this is purely `numba` CPU code and not CUDA

Reply:
I took a look again. My bad!

Reply:
@seanlaw 
Unfortunately, I am getting `typing` error.  It says: 
```
NumbaNotImplementedError: only one advanced index supported
```

Also see: [Numba Doc: Array Access](https://numba.pydata.org/numba-doc/dev/reference/numpysupported.html#array-access)

According to this [Numba Issue](https://github.com/numba/numba/issues/7869), when there is an array indexing, the integer index also counts as an advanced index, and it is not supported by Numba.

Reply:
@seanlaw 
Since I couldn't find `np.flatnonzero` in Numba Doc, I tried something similar before: `mask = np.nonzero(...)[0]` and it did not work (I got `typing` error and same message)

To make sure I am not missing anything, I just tried your recent proposal, and I got the same `typing` error and the same message.

Reply:
I think regardless of the representation of the mask (either by bool or int), we are dealing with `advanced indexing issue`.

Reply:
Sounds good. Thanks for trying